rocketship/dce/main.py

```diff
@@ -0,0 +1,271 @@
+"""
+Main DCE-MRI analysis functions for ROCKETSHIP.
+"""
+
+import os
+import numpy as np
+from typing import Dict, List, Optional, Tuple, Any
+import logging
+
+from ..utils.config import load_dce_preferences
+from ..utils.file_io import ImageLoader, VolumeLoader, FileParser
+from ..utils.image_processing import ImageProcessor, ROIAnalyzer
+from .models import ToftsModel, ExtendedToftsModel, PatlakModel, AIFModel
+
+
+logger = logging.getLogger(__name__)
+
+
+class DCEAnalyzer:
+    """Main DCE-MRI analysis class."""
+    
+    def __init__(self, config_file: Optional[str] = None):
+        """Initialize DCE analyzer with configuration."""
+        self.config = load_dce_preferences(config_file)
+        self.models = {
+            'tofts': ToftsModel(),
+            'extended_tofts': ExtendedToftsModel(),
+            'patlak': PatlakModel()
+        }
+    
+    def analyze_dce_data(self, 
+                        input_files: List[str],
+                        output_dir: str,
+                        aif_roi: Optional[np.ndarray] = None,
+                        model_type: str = 'tofts',
+                        verbose: bool = False) -> Dict[str, Any]:
+        """
+        Run complete DCE-MRI analysis.
+        
+        Args:
+            input_files: List of DCE time series files
+            output_dir: Output directory for results
+            aif_roi: ROI mask for AIF extraction (optional)
+            model_type: Pharmacokinetic model to use
+            verbose: Verbose output
+            
+        Returns:
+            Analysis results dictionary
+        """
+        logger.info(f"Starting DCE analysis with {len(input_files)} files")
+        
+        # Create output directory
+        os.makedirs(output_dir, exist_ok=True)
+        
+        # Load DCE time series
+        if verbose:
+            print("Loading DCE time series...")
+        
+        volume, metadata = VolumeLoader.load_image_series(input_files)
+        
+        if verbose:
+            print(f"Loaded volume shape: {volume.shape}")
+        
+        # Extract AIF
+        if aif_roi is not None:
+            aif_signal = ROIAnalyzer.extract_roi_signal(volume, aif_roi)
+        else:
+            # Auto-detect AIF (simplified version)
+            aif_signal = self._auto_detect_aif(volume, verbose=verbose)
```

Comment:
Most of the time this should supplied as a filepath to an .nii.gz ROI created by another repo (vascular_function), thus it should be skipped if the file exists/array was already extracted due to masking DCE with ROI.